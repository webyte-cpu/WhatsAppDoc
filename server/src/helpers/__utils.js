import objectKeysToCamelCase from "./objectKeyCase.js";
import objectFilter from "./objectFilter.js";
import { constructTestServer, cleanDb ,} from "./integrationTesting.js";

export {
  constructTestServer,
  objectKeysToCamelCase,
  objectFilter,
  cleanDb,
};
