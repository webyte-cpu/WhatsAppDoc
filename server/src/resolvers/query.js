import admin from "../models/admin.js";
import doctor from "../models/doctor.js";
import patient from "../models/patient.js";
import user from "../models/user.js";

const query = {
  admin: () => admin.get(),
  user: () => user.get(),
  doctor: () => doctor.get(),
  // patient: () => patient.get(),
};

export default query;
