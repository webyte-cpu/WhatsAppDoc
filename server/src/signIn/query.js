import { gql } from "apollo-server-express";

export const SIGN_IN = gql`
  mutation SignIn($email: EmailAddress!, $password: String!) {
    signIn(email: $email, password: $password)
  }
`;
