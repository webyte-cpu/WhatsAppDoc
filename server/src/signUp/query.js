import { gql } from "apollo-server-express";

export const SIGN_UP = gql`
  mutation SignUp(
    $firstName: String!
    $middleName: String
    $lastName: String!
    $email: EmailAddress!
    $password: String!
    $role: Role!
    $img: String
    $sex: Sex!
    $birthdate: Date!
    $doctor: DoctorInput
  ) {
    signUp(
      firstName: $firstName
      middleName: $middleName
      lastName: $lastName
      email: $email
      password: $password
      role: $role
      img: $img
      sex: $sex
      birthdate: $birthdate
      doctor: $doctor
    )
  }
`;
