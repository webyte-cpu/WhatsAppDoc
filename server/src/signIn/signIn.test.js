import { cleanDb, constructTestServer } from "../helpers/__utils.js";
import { createTestClient } from "apollo-server-testing";
import jwt from "jsonwebtoken";
import { SIGN_IN } from "./query.js";
import pg from "../../db/index.js";
describe("Sign In", () => {
  beforeAll(async () => {
    try {
      await cleanDb();
      await pg.seed.run();
    } catch (error) {
      console.log(error);
    }
  });

  afterAll(async () => {
    try {
      await cleanDb();
      await pg.destroy();
    } catch (error) {
      console.log(error);
    }
  });

  const { server } = constructTestServer();
  const { query } = createTestClient(server);

  it("Has an invalid email and password", async () => {
    const variables = { email: "", password: "" };

    const response = await query({
      mutation: SIGN_IN,
      variables,
    });

    await expect(response).toMatchSnapshot();
  });

  it("Has valid email but wrong password", async () => {
    const variables = { email: "kent@webyte.org", password: "asd" };

    const response = await query({
      mutation: SIGN_IN,
      variables,
    });

    await expect(response).toMatchSnapshot();
  });

  it("Trys to login a non-existing user", async () => {
    const variables = { email: "George@webyte.org", password: "L3tM3!N" };

    const response = await query({
      mutation: SIGN_IN,
      variables,
    });

    await expect(response).toMatchSnapshot();
  });

  it("Should login", async () => {
    const variables = { email: "kent@webyte.org", password: "$3cR3tP4ss2" };

    const response = await query({
      mutation: SIGN_IN,
      variables,
    });

    const { email, firstName, lastName, role, uid, verificationStatus } =
      jwt.verify(response.data.signIn, process.env.JWT_SECRET_KEY);

    await expect({
      email,
      firstName,
      lastName,
      role,
      uid,
      verificationStatus,
    }).toMatchSnapshot();
  });
});
