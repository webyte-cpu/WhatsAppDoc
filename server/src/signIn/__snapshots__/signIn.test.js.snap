// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In Has an invalid email and password 1`] = `
Object {
  "data": undefined,
  "errors": Array [
    [UserInputError: Variable "$email" got invalid value ""; Expected type "EmailAddress". Value is not a valid email address: ],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Sign In Has valid email but wrong password 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Invalid Email or Password.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Sign In Should login 1`] = `
Object {
  "email": "kent@webyte.org",
  "firstName": "Kent",
  "lastName": "Handumon",
  "role": "DOCTOR",
  "uid": null,
  "verificationStatus": "PENDING",
}
`;

exports[`Sign In Trys to login a non-existing user 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Invalid Email or Password.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
